src/paulscode/android/mupen64plusae/input/provider/AbstractProvider.java

```diff
@@ -206,6 +244,34 @@ public static String getHardwareName( int id )
     }
     
     /**
+     * Gets the unique name of the device.
+     * 
+     * @param id The unique hardware identifier.
+     * 
+     * @return The unique name of the device, or null if hardware not found.
+     */
+
+    @SuppressLint("NewApi")
+	public static String getUniqueName( int id )
+    {
+        if( id > HARDWARE_ID_MOGA_OFFSET && id <= HARDWARE_ID_MOGA_MAX )
+        {
+            return mogaPrefix + ( id - HARDWARE_ID_MOGA_OFFSET );
```

Comment:
Just FYI, I did test with a Moga Power Pro in A mode to make sure I hadn't completely messed something up. Somebody should probably still test it though.